aips/on-chain-randomness-impl.md

```diff
@@ -148,11 +127,211 @@ In each epoch, the validators will collaborate to produce randomness for *every*
 
 Importantly, only 50% or more of the stake can compute the wVRF evaluation, which ensures **unpredictability**. Furthermore, the uniqueness property of wVRFs together with the secrecy of the wPVSS scheme ensure **unbiasability** against adversaries controlling less than 50% of the stake.
 
-## Reference Implementation
+#### Consumption in user transactions
+
+The randomness seed of a block will be published on chain in a new `BlockMetadataExt` transaction that goes before any user transaction within the same block.
+A user transaction then combines the block-level seed and its unique transaction metadata to obtain the transaction-level randomness seed.
+
+#### How to land: new internal transaction type: `BlockMetadataExt`
+The wVRF evaluation of needs to be put on chain as the block randomness seed in the `BlockMetadata` transaction (always the 1st transction in a block).
+To be backward-compatible, a new transaction type `BlockMetadataExt` is needed (and updating `BlockMetadata` transaction is impossible).
+```rust
+pub enum Transaction { // the internal transaction type
+    UserTransaction(SignedTransaction),
+    GenesisTransaction(WriteSetPayload),
+    BlockMetadata(BlockMetadata),
+    StateCheckpoint(HashValue),
+    ValidatorTransaction(ValidatorTransaction),
+    BlockMetadataExt(BlockMetadataExt), // new variant
```

Comment:
how will the chain know to use this new variant and then to switch to v1?

Reply:
it's feature gated [here](https://github.com/aptos-labs/aptos-core/blob/1de391c3589cf2d07cb423a73c4a2a6caa299ebf/consensus/src/state_computer.rs#L203), also called out in a later section